src/field_effect.c

```diff
@@ -207,3 +613,2790 @@ bool8 FieldEffectActiveListContains(u8 id)
             return TRUE;
     return FALSE;
 }
+
+u8 CreateTrainerSprite_BirchSpeech(u8 gender, s16 x, s16 y, u8 subpriority, u8 *buffer)
+{
+    struct SpriteTemplate spriteTemplate;
+    LoadCompressedObjectPaletteOverrideBuffer(&gTrainerFrontPicPaletteTable[gender], buffer);
+    LoadCompressedObjectPicOverrideBuffer(&gTrainerFrontPicTable[gender], buffer);
+    spriteTemplate.tileTag = gTrainerFrontPicTable[gender].tag;
+    spriteTemplate.paletteTag = gTrainerFrontPicPaletteTable[gender].tag;
+    spriteTemplate.oam = &gOamData_839F0F4;
+    spriteTemplate.anims = gDummySpriteAnimTable;
+    spriteTemplate.images = NULL;
+    spriteTemplate.affineAnims = gDummySpriteAffineAnimTable;
+    spriteTemplate.callback = SpriteCallbackDummy;
+    return CreateSprite(&spriteTemplate, x, y, subpriority);
+}
+
+void LoadTrainerGfx_TrainerCard(u8 gender, u16 palOffset, u8 *dest)
+{
+    LZDecompressVram(gTrainerFrontPicTable[gender].data, dest);
+    LoadCompressedPalette(gTrainerFrontPicPaletteTable[gender].data, palOffset, 0x20);
+}
+
+u8 CreateBirchSprite(s16 x, s16 y, u8 subpriority)
+{
+    LoadSpritePalette(&gUnknown_0839F114);
+    return CreateSprite(&gSpriteTemplate_839F128, x, y, subpriority);
+}
+
+u8 CreateMonSprite_PicBox(u16 species, s16 x, s16 y, u8 subpriority)
+{
+    DecompressPicFromTable_2(&gMonFrontPicTable[species], gMonFrontPicCoords[species].coords, gMonFrontPicCoords[species].y_offset, gUnknown_081FAF4C[3], gUnknown_081FAF4C[3], species);
+    LoadCompressedObjectPalette(&gMonPaletteTable[species]);
+    GetMonSpriteTemplate_803C56C(species, 3);
+    gUnknown_02024E8C.paletteTag = gMonPaletteTable[0].tag;
+    sub_807DE38(IndexOfSpritePaletteTag(gMonPaletteTable[0].tag) + 0x10);
+    return CreateSprite(&gUnknown_02024E8C, x, y, subpriority);
+}
+
+u8 CreateMonSprite_FieldMove(u16 species, u32 d, u32 g, s16 x, s16 y, u8 subpriority)
+{
+    const struct SpritePalette *spritePalette;
+    HandleLoadSpecialPokePic(&gMonFrontPicTable[species], gMonFrontPicCoords[species].coords, gMonFrontPicCoords[species].y_offset, (u32)gUnknown_081FAF4C[3] /* this is actually u8* or something, pointing to ewram */, gUnknown_081FAF4C[3], species, g);
+    spritePalette = sub_80409C8(species, d, g);
+    LoadCompressedObjectPalette(spritePalette);
+    GetMonSpriteTemplate_803C56C(species, 3);
+    gUnknown_02024E8C.paletteTag = spritePalette->tag;
+    sub_807DE38(IndexOfSpritePaletteTag(spritePalette->tag) + 0x10);
+    return CreateSprite(&gUnknown_02024E8C, x, y, subpriority);
+}
+
+void FreeResourcesAndDestroySprite(struct Sprite *sprite)
+{
+    sub_807DE68();
+    FreeSpritePaletteByTag(GetSpritePaletteTagByPaletteNum(sprite->oam.paletteNum));
+    if (sprite->oam.affineMode != 0)
+    {
+        FreeOamMatrix(sprite->oam.matrixNum);
+    }
+    DestroySprite(sprite);
+}
+
+#undef NONMATCHING
```

Comment:
idk i thought i saw `#undef NONMATCHING` somewhere else in source. at any rate, this comment is obsoleted by d8f4d3da44d52ac56d93a41edb24cf0a803f5c67